theme/views.py

```diff
@@ -156,3 +168,122 @@ def signup(request, template="accounts/account_signup.html"):
+                    info(request, _("A verification email has been sent to your new email with "
+                                    "a link for updating your email. If you "
+                                    "do not receive this email please check your "
+                                    "spam folder as sometimes the confirmation email "
+                                    "gets flagged as spam. If you entered an incorrect "
+                                    "email address, please request email update again. "
+                                    ))
+                    # send an email to the old address notifying the email change
+                    send_mail(subject="Your HydroShare email.",
+                              message="You have changed your HydroShare account email",
+                              from_email= settings.DEFAULT_FROM_EMAIL, recipient_list=[old_email], fail_silently=False)
+            else:
+                errors = {}
+                if not user_form.is_valid():
+                    errors.update(user_form.errors)
+
+                if not profile_form.is_valid():
+                    errors.update(profile_form.errors)
+
+                msg = ' '.join([err[0] for err in errors.values()])
+                messages.error(request, msg)
+
+    except Exception as ex:
+        messages.error(request, ex.message)
+
+    return HttpResponseRedirect(request.META['HTTP_REFERER'])
+
+
+def send_verification_mail_for_email_update(request, user, new_email, verification_type):
+    """
+    Sends an email with a verification link to users when
+    they update their email. The email is sent to the new email.
+    The actual update of the email happens only after
+    the verification link is clicked.
+    The ``verification_type`` arg is both the name of the urlpattern for
+    the verification link, as well as the names of the email templates
+    to use.
+    """
+    verify_url = reverse(verification_type, kwargs={
+        "uidb36": int_to_base36(user.id),
+        "token": default_token_generator.make_token(user),
+        "new_email": new_email
+    }) + "?next=" + (next_url(request) or "/")
+    context = {
+        "request": request,
+        "user": user,
+        "new_email": new_email,
+        "verify_url": verify_url,
+    }
+    subject_template_name = "email/%s_subject.txt" % verification_type
+    subject = subject_template(subject_template_name, context)
+    send_mail_template(subject, "email/%s" % verification_type,
+                       settings.DEFAULT_FROM_EMAIL, new_email,
+                       context=context)
+
+
+def email_verify(request, new_email, uidb36=None, token=None):
+    """
+    View for the link in the verification email sent to a user
+    when they update their email as part of profile update.
+    User email is set to new_email and logs them in,
+    redirecting to the URL they tried to access when profile.
```

Comment:
@pkdash @mauriel133 The last two words "when profile" are not clear, perhaps change it to something like "when changing profile" or something like that?